File: Heat/ViewModelBuilders/SerialSchemeViewModelBuilder.vb
Imports Heat.Models
Imports AutoMapper

Public Class SerialSchemeViewModelBuilder

    Private _db As IHeatDBContext

    Public Sub New(context As IHeatDBContext)
        If IsNothing(context) Then
            Throw New Exception("Context!")
        End If

        _db = context
    End Sub

    Public Function getListViewModel() As List(Of IndexSerialSchemeViewModel)
        Dim result As New List(Of IndexSerialSchemeViewModel)

        Dim l As New List(Of SerialScheme)
        l = _db.SerialSchemes.ToList

        result = Mapper.Map(Of List(Of SerialScheme), List(Of IndexSerialSchemeViewModel))(l)

        Return result


    End Function

End Class

File: Heat.ConvertedToC#/ViewModelBuilders/SerialSchemeViewModelBuilder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using Heat.Models;
using AutoMapper;
namespace Heat
{

    public class SerialSchemeViewModelBuilder
	{


		private IHeatDBContext _db;
		public SerialSchemeViewModelBuilder(IHeatDBContext context)
		{
			if ((context == null)) {
				throw new Exception("Context!");
			}

			_db = context;
		}

		public List<IndexSerialSchemeViewModel> getListViewModel()
		{
			List<IndexSerialSchemeViewModel> result = new List<IndexSerialSchemeViewModel>();

			List<SerialScheme> l = new List<SerialScheme>();
			l = _db.SerialSchemes.ToList();

			result = Mapper.Map<List<SerialScheme>, List<IndexSerialSchemeViewModel>>(l);

			return result;


		}

	}
}
